### ISSUE TITLE
SQLITE Error: no such column

### ISSUE THREAD

**Issue by RyoshiKayo** [WeebDev/lolisafe#91]
On upload from the extension to an album I get the following error in the lolisafe console:
```
{ Error: SQLITE_ERROR: no such column: editedAt errno: 1, code: 'SQLITE_ERROR' }
```

I'm not sure if they are related but I got this error at the same time
```
Unhandled rejection Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at validateHeader (_http_outgoing.js:503:11)
    at ServerResponse.setHeader (_http_outgoing.js:510:3)
    at ServerResponse.header (/home/kayo/lolisafe/node_modules/express/lib/response.js:767:10)
    at ServerResponse.send (/home/kayo/lolisafe/node_modules/express/lib/response.js:170:12)
    at ServerResponse.send (/home/kayo/lolisafe/node_modules/express-debug/lib/response.js:124:29)
    at ServerResponse.json (/home/kayo/lolisafe/node_modules/express/lib/response.js:267:15)
    at db.table.where.update.then.catch.error (/home/kayo/lolisafe/controllers/uploadController.js:160:47)
    at tryCatcher (/home/kayo/lolisafe/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/kayo/lolisafe/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/kayo/lolisafe/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/kayo/lolisafe/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/kayo/lolisafe/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/home/kayo/lolisafe/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/kayo/lolisafe/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/kayo/lolisafe/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:763:18)
```

Currently trying to reproduce the issue.

**Comment by RyoshiKayo** [comment]
This happens when you upload a new image to an album. This doesn't happen if I just "send to safe".
I also tested by using the frontpage and get the same errors.

Here is the full output from ```DEBUG=* node lolisafe.js```
```
  express:router dispatching POST /api/upload/42 +15s
  express:router query  : /api/upload/42 +0ms
  express:router expressInit  : /api/upload/42 +1ms
  express:router EDT  : /api/upload/42 +0ms
  express:router helmet  : /api/upload/42 +0ms
  express:router urlencodedParser  : /api/upload/42 +1ms
  body-parser:urlencoded content-type "multipart/form-data; boundary=----WebKitFormBoundary59LLdO6W7kAst55j" +0ms
  body-parser:urlencoded skip parsing +0ms
  express:router jsonParser  : /api/upload/42 +0ms
  body-parser:json content-type "multipart/form-data; boundary=----WebKitFormBoundary59LLdO6W7kAst55j" +0ms
  body-parser:json skip parsing +0ms
  express:router serveStatic  : /api/upload/42 +0ms
  express:router router  : /api/upload/42 +0ms
  express:router dispatching POST /api/upload/42 +0ms
  express:router trim prefix (/api) from url /api/upload/42 +1ms
  express:router router /api : /api/upload/42 +0ms
  express:router dispatching POST /upload/42 +0ms
  knex:client acquired connection from pool: __knexUid2 +15s
  knex:query select * from `users` where `token` = ? limit ? +15s
  knex:bindings [ '<token>',
  knex:bindings   1 ] +15s
  knex:client releasing connection to pool: __knexUid2 +1ms
  knex:client acquired connection from pool: __knexUid3 +0ms
  knex:query select * from `users` where `token` = ? limit ? +2ms
  knex:bindings [ '<token>',
  knex:bindings   1 ] +2ms
  knex:client releasing connection to pool: __knexUid3 +1ms
  knex:client acquired connection from pool: __knexUid3 +1ms
  knex:query select * from `albums` where `id` = ? and `userid` = ? limit ? +1ms
  knex:bindings [ '42', 1, 1 ] +1ms
  knex:client releasing connection to pool: __knexUid3 +1ms
  knex:client acquired connection from pool: __knexUid3 +5ms
  knex:query select * from `files` where (`userid` = ?) and `hash` = ? and `size` = ? limit ? +6ms
  knex:bindings [ 1, '<token>', 83583, 1 ] +6ms
  knex:client releasing connection to pool: __knexUid3 +1ms
  knex:client acquired connection from pool: __knexUid3 +1ms
  knex:query insert into `files` (`albumid`, `hash`, `ip`, `name`, `original`, `size`, `timestamp`, `type`, `userid`) values (?, ?, ?, ?, ?, ?, ?, ?, ?) +3ms
  knex:bindings [ '42',
  knex:bindings   '97a839c5c90003ffac4108912997f61d',
  knex:bindings   '<my-ip>',
  knex:bindings   'MLlPc5E4.png',
  knex:bindings   'upload.png',
  knex:bindings   83583,
  knex:bindings   1522842125,
  knex:bindings   'image/png',
  knex:bindings   1 ] +3ms
  knex:client releasing connection to pool: __knexUid3 +117ms
  knex:client acquired connection from pool: __knexUid3 +2ms
  knex:query update `albums` set `editedAt` = ? where `id` = ? +119ms
  knex:bindings [ 1522842125, '42' ] +119ms
  knex:client releasing connection to pool: __knexUid3 +3ms
{ Error: SQLITE_ERROR: no such column: editedAt errno: 1, code: 'SQLITE_ERROR' }
Unhandled rejection Error [ERR_HTTP_HEADERS_SENT]: Cannot set headers after they are sent to the client
    at validateHeader (_http_outgoing.js:503:11)
    at ServerResponse.setHeader (_http_outgoing.js:510:3)
    at ServerResponse.header (/home/kayo/lolisafe/node_modules/express/lib/response.js:767:10)
    at ServerResponse.send (/home/kayo/lolisafe/node_modules/express/lib/response.js:170:12)
    at ServerResponse.send (/home/kayo/lolisafe/node_modules/express-debug/lib/response.js:124:29)
    at ServerResponse.json (/home/kayo/lolisafe/node_modules/express/lib/response.js:267:15)
    at db.table.where.update.then.catch.error (/home/kayo/lolisafe/controllers/uploadController.js:160:47)
    at tryCatcher (/home/kayo/lolisafe/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/home/kayo/lolisafe/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/home/kayo/lolisafe/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/home/kayo/lolisafe/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/home/kayo/lolisafe/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/home/kayo/lolisafe/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/home/kayo/lolisafe/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/kayo/lolisafe/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:763:18)
```

If I upload the same image again to the same album I get the following:
```
  express:router dispatching POST /api/upload/42 +4m
  express:router query  : /api/upload/42 +0ms
  express:router expressInit  : /api/upload/42 +0ms
  express:router EDT  : /api/upload/42 +1ms
  express:router helmet  : /api/upload/42 +0ms
  express:router urlencodedParser  : /api/upload/42 +0ms
  body-parser:urlencoded content-type "multipart/form-data; boundary=----WebKitFormBoundaryrC6qRSnjuCTkiEPw" +0ms
  body-parser:urlencoded skip parsing +0ms
  express:router jsonParser  : /api/upload/42 +0ms
  body-parser:json content-type "multipart/form-data; boundary=----WebKitFormBoundaryrC6qRSnjuCTkiEPw" +0ms
  body-parser:json skip parsing +0ms
  express:router serveStatic  : /api/upload/42 +0ms
  express:router router  : /api/upload/42 +1ms
  express:router dispatching POST /api/upload/42 +0ms
  express:router trim prefix (/api) from url /api/upload/42 +0ms
  express:router router /api : /api/upload/42 +0ms
  express:router dispatching POST /upload/42 +0ms
  knex:client acquired connection from pool: __knexUid2 +4m
  knex:query select * from `users` where `token` = ? limit ? +4m
  knex:bindings [ '<token>',
  knex:bindings   1 ] +4m
  knex:client releasing connection to pool: __knexUid2 +1ms
  knex:client acquired connection from pool: __knexUid3 +0ms
  knex:query select * from `users` where `token` = ? limit ? +2ms
  knex:bindings [ '<token>',
  knex:bindings   1 ] +2ms
  knex:client releasing connection to pool: __knexUid3 +1ms
  knex:client acquired connection from pool: __knexUid3 +1ms
  knex:query select * from `albums` where `id` = ? and `userid` = ? limit ? +1ms
  knex:bindings [ '42', 1, 1 ] +1ms
  knex:client releasing connection to pool: __knexUid3 +1ms
  knex:client acquired connection from pool: __knexUid3 +4ms
  knex:query select * from `files` where (`userid` = ?) and `hash` = ? and `size` = ? limit ? +5ms
  knex:bindings [ 1, '97a839c5c90003ffac4108912997f61d', 83583, 1 ] +5ms
  knex:client releasing connection to pool: __knexUid3 +1ms
{ Error: ENOENT: no such file or directory, stat '/home/kayo/lolisafe/uploads/thumbs/4Cc1wO0M.png'
  errno: -2,
  code: 'ENOENT',
  syscall: 'stat',
  path: '/home/kayo/lolisafe/uploads/thumbs/4Cc1wO0M.png' }
```

**Comment by BobbyWibowo** [comment]
For the no columns, you will have to run "node database/migration.js" once (it is odd, I know, they forgot to add the column names to "database/db.js" (if someone is free please make a PR - I already got this solved on my end long ago but I kinda forgotten it ever happened).

"Cannot set headers after they are sent" happened because "res.json()" or the likes are being called more than once in a single session. I don't recall ever getting this during the time I used to NOT have the columns, but there were indeed some other situations where this used to happen on my end.

**Comment by RyoshiKayo** [comment]
I need to run the migration script when this was a fresh install from a couple weeks ago? 🤔

**Comment by Pitu** [comment]
Fixed on 91331e794788f3a8585fa90912cd3dd3bd6e9f22

**Comment by RyoshiKayo** [comment]
Ran the script none the less and it has fixed the issue 👌
